kube-aws-updater

```diff
@@ -129,77 +130,118 @@ label_for_cycling() {
+    jq -r '.Reservations[].Instances[].InstanceId')
+  local asg_name=$(aws ${aws_opts} --output=json autoscaling describe-auto-scaling-groups |\
+    jq -r '."AutoScalingGroups"[] | select(."Instances"[]."InstanceId"=="'${instance_id}'") | ."AutoScalingGroupName"')
+  local asg_count=$(aws ${aws_opts} --output=json autoscaling describe-auto-scaling-groups |\
+    jq -r '."AutoScalingGroups"[] | select(."Instances"[]."InstanceId"=="'${instance_id}'") | ."DesiredCapacity"')
+
+  aws ${aws_opts} --output=json autoscaling update-auto-scaling-group \
+    --auto-scaling-group-name "${asg_name}" \
+    --desired-capacity $(( ${asg_count} * 2)) \
+    --max-size $(( ${asg_count} * 2))
 
   # - wait for 2x group
   local count=0
   while [[ ${count} -lt ${asg_count} ]]; do
     sleep 8
-    count=$(kubectl --context=${kube_context} get node -lrole=${role} -Lrole,retiring \
-      | grep -v ${retire_time} \
-      | grep -v NotReady \
-      | tail -n +2 \
-      | wc -l)
+    count=$(kubectl --context=${kube_context} get node -lrole=${role} -Lrole,retiring |\
+      grep -v ${retire_time} |\
+      grep -v NotReady |\
+      tail -n +2 |\
+      wc -l)
     echo "${count}/${asg_count} ready ${role}s"
   done
 
   # - stop ASG actions
-  aws ${aws_opts} --output=json autoscaling suspend-processes --auto-scaling-group-name "${asg_name}"
+  aws ${aws_opts} --output=json autoscaling suspend-processes \
+    --auto-scaling-group-name "${asg_name}" --scaling-processes \
+    "Launch" \
+    "ReplaceUnhealthy" \
+    "AZRebalance" \
+    "AlarmNotification" \
+    "ScheduledActions" \
+    "AddToLoadBalancer"
 
   ### - move the registry
   if [[ "${role}" == "worker" ]]; then
     handle_registry
   fi
 
-  # - drain/terminate labelled masters (sequentially)
+  # - drain/terminate labelled nodes (sequentially)
   local old_nodes=$(kubectl --context="${kube_context}" get nodes -l role="${role}",retiring="${retire_time}" -o json | jq -r '.items[].metadata.name')
   for old_node in ${old_nodes}; do
     kill_node ${old_node}
   done
 
   # - resize ASG to original size
-  aws ${aws_opts} --output=json autoscaling update-auto-scaling-group --auto-scaling-group-name "${asg_name}" --desired-capacity ${asg_count} --max-size ${asg_count}
+  aws ${aws_opts} --output=json autoscaling update-auto-scaling-group \
+    --auto-scaling-group-name "${asg_name}" --desired-capacity ${asg_count} \
+    --max-size ${asg_count}
+
+  # sleep some time to allow ASG to catch and see the "Terminating" instance
+  sleep 64
+
+  # wait for ASG to catch up and have no instances "Terminating"
+  set +e
+  local rc=1
+  while [[ ${rc} -ne 4 ]]; do
+    sleep 16
+    aws ${aws_opts} --output=json autoscaling describe-auto-scaling-groups \
+      --auto-scaling-group-names "${asg_name}" |\
+      jq -e -r '."AutoScalingGroups"[0]."Instances"[] | select (."LifecycleState" == "Terminating")'
```

Comment:
I am thinking if that would better be something like:
```
aws ${aws_opts} --output=json autoscaling describe-auto-scaling-groups --auto-scaling-group-names "${asg_name}" | jq -e '."AutoScalingGroups"[0]."Instances" | length'
```
and then compare it with `${asg_count}` (maybe in the while loop if that oneliner doesn't look too bad). 
I am just a bit concerned if exit code 4 is something meaningful here.
```
-e / --exit-status:

Sets the exit status of jq to 0 if the last output values was neither false nor null, 1 if the last output value was either false or null, or 4 if no valid result was ever produced
```
from https://stedolan.github.io/jq/manual/